src/test/java/com/google/api/generator/engine/writer/JavaWriterVisitorTest.java

```diff
@@ -73,6 +76,12 @@ public void writePrimitiveArrayType() {
 
   /** =============================== EXPRESSIONS =============================== */
   @Test
+  public void writeStringObjectValue() {
+    StringObjectValue s = StringObjectValue.builder().setValue("test").build();
+    assertThat(s.value()).isEqualTo("test");
+    assertThat(s.type()).isEqualTo(TypeNode.STRING);
+  }
```

Comment:
Add an `AssignmentExpr` test so we can see this in larger context?

Reply:
thx for the suggestion, working on it!

Reply:
this is very helpful! turns out the class needs a `toString()` method to return just the value(), else it will use the default toString() which will fail the assignment expression!